### PR TITLE
Enable SmoothQuant only for OV backend

### PR DESCRIPTION
**Pull request by andrey-churkin**
### Changes

Run the SmoothQuant and ChannelAlignment algorithms only for the OpenVINO backend.

### Reason for changes

Only the OpenVINO backend supports SmoothQuant and ChannelAlignment algorithms.

### Related tickets
N/A

### Tests
N/A

### CODE REVIEW

File: nncf/quantization/algorithms/channel_alignment/algorithm.py [hunk 1]
```diff
@@ -97,6 +97,10 @@ def apply(
         statistic_points: Optional[StatisticPointsContainer] = None,
         dataset: Optional[Dataset] = None,
     ) -> TModel:
+        backend = get_backend(model)
+        if backend != BackendType.OPENVINO:
+            return model
```
Comment (daniil-lyakhov) [inline nncf/quantization/algorithms/channel_alignment/algorithm.py:102]: Should we add a warning as it was before?

**Review by alexsu52 — CHANGES_REQUESTED** [review]
Could check it outside of the algorithm? For example you can use available_backends.

**Comment by alexsu52** [comment]
> > Could check it outside of the algorithm? For example you can use available_backends.
> 
> The backend is unknown when the pipeline is created.
> 
> https://github.com/openvinotoolkit/nncf/blob/2878672e66ad2353d831975187a1a44aa35108ba/nncf/quantization/algorithms/post_training/algorithm.py#L69-L71
> 
> One possible place where it can be handled is here.
> 
> https://github.com/openvinotoolkit/nncf/blob/2878672e66ad2353d831975187a1a44aa35108ba/nncf/quantization/algorithms/pipeline.py#L112-L115
> 
> We will skip the step if the backend is not in the available backends list. Please let me know if you are okay with this, or maybe you have another suggestion.

This solution is better.

File: nncf/quantization/algorithms/channel_alignment/algorithm.py (hunk 1 above)
Comment (andrey-churkin) [reply nncf/quantization/algorithms/channel_alignment/algorithm.py]: Done

**Review by daniil-lyakhov — APPROVED** [review]

**Review by alexsu52 — APPROVED** [review]

**Comment by andrey-churkin** [comment]
The `available_backends` property has an incorrect implementation; it does not work.
See the following example
```python
from nncf.quantization.algorithms.channel_alignment.algorithm import ChannelAlignment
ca = ChannelAlignment()
print(ca.available_backends)
# Out: {}
```
It is not possible to obtain available backends until you call the `apply(`) or `get_statistic_points()` methods.

**Pull request merged**